lib/fizzy/execute.hpp

```diff
@@ -22,7 +22,7 @@ struct Instance;
 
 struct ExternalFunction
 {
-    std::function<execution_result(Instance&, std::vector<uint64_t>)> function;
+    std::function<execution_result(Instance&, std::vector<uint64_t>, int depth)> function;
```

Comment:
I think this is not nice -- any way to avoid it?

Reply:
Another way I see is create another type alongside `ExternalFunction` which will be used only in the table and will have `depth`

Reply:
I mean this PR then will not be needed (`ExternalFunction` doesn't change), but then the solution in #246 will change from https://github.com/wasmx/fizzy/pull/246/files#diff-5292aa18dd21e020324623b99674f3da to something like
```
struct TableFunction
{
    std::function<execution_result(std::vector<uint64_t>, int depth)> function;
    FuncType type;
};
using table_elements = std::vector<std::optional<TableFunction>>;
```

Reply:
Here's this alternative https://github.com/wasmx/fizzy/pull/266
But I'm not sure if it's better.